hazelcast/src/main/java/com/hazelcast/concurrent/atomiclong/AtomicLongContainer.java

```diff
@@ -38,30 +53,58 @@ public long get() {
 
     public long addAndGet(long delta) {
         value += delta;
+        isModified = true;
         return value;
     }
 
     public void set(long value) {
         this.value = value;
+        isModified = true;
     }
 
     public boolean compareAndSet(long expect, long value) {
         if (this.value != expect) {
             return false;
         }
         this.value = value;
+        isModified = true;
         return true;
     }
 
     public long getAndAdd(long delta) {
         long tempValue = value;
         value += delta;
+        isModified = true;
         return tempValue;
     }
 
     public long getAndSet(long value) {
         long tempValue = this.value;
         this.value = value;
+        isModified = true;
         return tempValue;
     }
+
+    @Override
+    public Long merge(SplitBrainMergeEntryView<Void, Long> mergingEntry, SplitBrainMergePolicy mergePolicy) {
+        mergePolicy.setSerializationService(nodeEngine.getSerializationService());
+
+        if (isModified) {
+            SplitBrainMergeEntryView<Void, Long> existingEntry = createSplitBrainMergeEntryView(value);
+            Long newValue = mergePolicy.merge(mergingEntry, existingEntry);
+            if (newValue != null && !newValue.equals(value)) {
```

Comment:
Does the contract of the `SplitBrainMergePolicy#merge` method say what happens when the user returns `null`?

Reply:
I think I didn't specify this yet. At least one of the old implementations removes the whole entry, when `null` is returned (I think it was `IMap`). But especially for the key-less entries this is going to be hard. So I think I won't implement this behavior. The current state on `null` is, that the existing state will be preserved (no entry or unchanged existing entry).